lib/install/ova/configure.go

```diff
@@ -0,0 +1,123 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ova
+
+import (
+	"context"
+	"net"
+	"net/url"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/lib/config/dynamic/admiral"
+	"github.com/vmware/vic/pkg/errors"
+	"github.com/vmware/vic/pkg/vsphere/session"
+	"github.com/vmware/vic/pkg/vsphere/tasks"
+	"github.com/vmware/vic/pkg/vsphere/vm"
+)
+
+const (
+	// ManagedByKey defines the extension key to use in the ManagedByInfo of the OVA
+	ManagedByKey = "com.vmware.vic"
+	// ManagedByType defines the type to use in the ManagedByInfo of the OVA
+	ManagedByType = "VicApplianceVM"
+)
+
+func configureManagedByInfo(ctx context.Context, sess *session.Session, v *vm.VirtualMachine) error {
+	spec := types.VirtualMachineConfigSpec{
+		ManagedBy: &types.ManagedByInfo{
+			ExtensionKey: ManagedByKey,
+			Type:         ManagedByType,
+		},
+	}
+
+	info, err := v.WaitForResult(ctx, func(ctx context.Context) (tasks.Task, error) {
+		return v.Reconfigure(ctx, spec)
+	})
+
+	if err != nil {
+		log.Errorf("Error while setting ManagedByInfo: %s", err)
+		return err
+	}
+
+	if info.State != types.TaskInfoStateSuccess {
+		log.Errorf("Setting ManagedByInfo reported: %s", info.Error.LocalizedMessage)
+		return err
+	}
+
+	return nil
+}
+
+// ConfigureManagedByInfo takes sets the ManagedBy field for the VM specified by ovaURL
+func ConfigureManagedByInfo(ctx context.Context, sess *session.Session, ovaURL string) error {
+	v, err := getOvaVMByTag(ctx, sess, ovaURL)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Attempting to configure ManagedByInfo")
+	err = configureManagedByInfo(ctx, sess, v)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Successfully configured ManagedByInfo")
+	return nil
+}
+
+func getOvaVMByTag(ctx context.Context, sess *session.Session, u string) (*vm.VirtualMachine, error) {
+	ovaURL, err := url.Parse(u)
+	if err != nil {
+		return nil, err
+	}
+
+	host := ovaURL.Hostname()
+	ips, err := net.LookupIP(host)
+	if err != nil {
+		return nil, errors.Errorf("IP lookup failed: %s", err)
+	}
+
+	var ip string
+	for _, i := range ips {
+		if i.To4() != nil {
+			ip = i.String()
+		}
+	}
+
+	if ip == "" {
+		return nil, errors.Errorf("IPV6 support not yet implemented")
+	}
+
+	vms, err := admiral.DefaultDiscovery.Discover(ctx, sess)
+	if err != nil {
+		return nil, errors.Errorf("failed to discover OVA: %s", err)
+	}
+
+	for i, v := range vms {
+		vmIP, err := v.WaitForIP(ctx)
+		if err != nil && i == len(vms)-1 {
+			return nil, errors.Errorf("failed to get VM IP: %s", err)
+		}
+
+		// verify the tagged vm has the IP we expect
+		if vmIP == ip {
+			log.Debugf("Found OVA with matching IP: %s", ip)
```

Comment:
I wonder if we might want some more debug logging like this. (Otherwise, I could imagine this being hard to troubleshoot in a production environment.)

Maybe we could try logging the hostname that we're looking up, the IPs we find for it, the tagged vms admiral returns, and _then_ information about the one we found (maybe including some other information from `v`, like its name or moref, since we know IP isn't a unique identifier?)

Reply:
Fixed!